### PR TITLE
Problem: Jetstream and Cyverse ansible was becoming hard to maintain

### PR DESCRIPTION
**Pull request by steve-gregory**
*Solution: Move Jetstream's atmosphere-ansible to the Cyverse repo*

Goals:

- [x] Work with ansible-savvy team member to work through the differences between the branches (Thank you @c-mart !)
- [x] Use variables (+ `when:` clause) to allow ansible to run/skip sections of code which do not apply to jetstream and/or cyverse.

### CODE REVIEW

File: ansible/roles/atmo-fail2ban/vars/main.yml [hunk 1]
```diff
@@ -1,6 +1,9 @@
 SSH_PORT: '22'
 BANTIME: '1800'
-MAX_RETRY: '3'
```
Comment (c-mart) [inline ansible/roles/atmo-fail2ban/vars/main.yml]: Let's set these in group_vars per-environment and only leave reasonable defaults here?

File: ansible/roles/atmo-ntp/tasks/main.yml [hunk 2]
```diff
@@ -54,6 +54,7 @@
   shell: ntpdate 1.us.pool.ntp.org
   when: ntp_installed.stat.exists == False
   tags:
+    - ntp
```
Comment (c-mart) [inline ansible/roles/atmo-ntp/tasks/main.yml]: These are also not needed if we conditionally include this role.

File: ansible/roles/atmo-realvncserver/files/config.custom [hunk 3]
```diff
@@ -1,6 +1,8 @@
 # vncserver common.custom configuration.
 SecurityTypes=RA2
-Permissions=root:f,%core-services:f,:f
+#TODO: Line below CyVerse ONLY
```
Comment (c-mart) [inline ansible/roles/atmo-realvncserver/files/config.custom]: Perhaps make this file a template and allow the consumer to define this as a variable?

File: ansible/roles/atmo-setup-user/defaults/main.yml [hunk 4]
```diff
@@ -19,3 +19,9 @@ SUDOERS_USERS_AND_GROUPS:
     atmouser_nopass:
       line: "{{ ATMOUSERNAME }} ALL=(ALL) NOPASSWD:ALL"
       reg: "^{{ ATMOUSERNAME }}\\s+ALL=\\(ALL\\)\\s*NOPASSWD:ALL"
+# CYVERSE ONLY
```
Comment (c-mart) [inline ansible/roles/atmo-setup-user/defaults/main.yml]: Also could be defined in group_vars instead

File: ansible/roles/atmo-setup-user/tasks/main.yml [hunk 5]
```diff
@@ -10,31 +10,69 @@
...
+
 - name: add users and groups to sudoers
   lineinfile: dest=/etc/sudoers backup=yes regexp={{ item.value.reg }} line={{ item.value.line }} state=present
   with_dict: "{{ SUDOERS_USERS_AND_GROUPS.TO_ADD }}"
   tags: sudoers
 
 - name: start setting up /etc/skel/.bashrc
   lineinfile: dest=/etc/skel/.bashrc line='## Atmosphere System'
+  tags:
+    - cyverse-user-setup
```
Comment (c-mart) [inline ansible/roles/atmo-setup-user/tasks/main.yml]: I would instead split these into another tasks file and conditionally include them when someone sets the appropriate variable, ditto below for the tasks tagged "jetstream-user-setup".

File: ansible/roles/ldap/tasks/main.yml [hunk 6]
```diff
@@ -4,7 +4,9 @@
 - name: set python for centos-5
   set_fact: ansible_python_interpreter=/usr/bin/python
   when: ansible_distribution == "CentOS" and ansible_distribution_major_version == "5"
-  tags: python
+  tags:
+    - ldap
```
Comment (c-mart) [inline ansible/roles/ldap/tasks/main.yml]: these are also not needed if role is conditionally included

File: ansible/roles/mount-volume/defaults/main.yml [hunk 7]
```diff
@@ -1,2 +1,3 @@
 MOUNTNAME: "/vol"
 MOUNTTYPE: "ext3"
+# NOTE:ON JETSTREAM, MOUNTTYPE== ext4
```
Comment (c-mart) [inline ansible/roles/mount-volume/defaults/main.yml]: For a different PR, but should CyVerse also migrate to ext4 volumes?

**Comment by c-mart** [comment]
[My PR to this PR](https://github.com/cyverse/atmosphere-ansible/pull/61) implements the changes in the code review above. Still for us to do:

- [ ] Resolve some questions in my PR #150 regarding atmo-setup-user role
- [ ] Populate atmo-common role with deployment-specific motd templates
- Make appropriate changes in group_vars of our environment-specific repos to call appropriate optional roles and set other-specific behavior (I have a draft of this locally)
  - [ ] atmo-prod
  - [ ] atmobeta
  - [ ] atmo-dev
  - [ ] atmo-local?
  - [ ] Jetstream secrets
  - [ ] Atmosphere for Jetstream test cloud

Also, I haven't tried running any of this yet! Here is a proposed testing plan before merging to master (or deploying to production).

Environments we'll need to test against:
- [ ] Deployment with CyVerse's featured images (atmobeta)
- [ ] Deployment with at least one CentOS 7 image (Jetstream test cloud?)
- [ ] Deployment with Jetstream Cloud's featured images (Copy them to Jetstream test cloud?)

Things I think we should ensure for the relevant images each test environment:
- Optional playbooks run as appropriate per environment
- atmo-dhcp role does what we expect
- fail2ban role configures iptables and fail2ban correctly for all supported distros including CentOS 7; fail2ban it bans things when it should
- RealVNC server is configured with environment-specific permissions
- atmo-setup-user role configures local accounts as expected (??) for each environment
  - password-required sudo for user in CyVerse, passwordless or password-required sudo for user on Jetstream cloud?
- /home and cinder volumes mount and unmount as expected

**Comment by steve-gregory** [comment]
Additional testing will be tracked in our internal JIRA: [ATMO-1700](https://pods.iplantcollaborative.org/jira/browse/ATMO-1700)

**Pull request merged**